### PR TITLE
Deleted x_derivative functions, except for entropy. Also deleted ther…

### PR DESCRIPTION
**Pull request by moorek**
…mal_expansion_coeff implementation in interface.cc

### CODE REVIEW

**Comment by tjhei** [comment]
Let's see what the tester says: /run-tests

**Comment by bangerth** [comment]
Are any of the removed functions defined in the concrete implementations of material models?

**Comment by moorek** [comment]
I looked through the .cc files. The removed derivatives don't appear to be implemented.

The removed thermal_expansion_coefficient(   ) function (removed from the .cc but not the .h) is implemented in some material models, yes.

**Comment by tjhei** [comment]
> I looked through the .cc files. The removed derivatives don't appear to be implemented.

good.

> The removed thermal_expansion_coefficient( ) function (removed from the .cc but not the .h) is implemented in some material models, yes.

Yes, that is as intended.

File: include/aspect/material_model/interface.h [hunk 1]
```diff
@@ -581,110 +575,6 @@ namespace aspect
          */
```
Comment (jdannberg) [inline include/aspect/material_model/interface.h]: This comment about the partial derivatives can be removed as well.

**Comment by tjhei** [comment]
Looks good to me and I couldn't find any other place that uses these functions (benchmarks, cookbooks, tests, etc..).

**Pull request merged**